### DISCUSSION TITLE
Helper function to locate fixture path?

### DISCUSSION THREAD

**Discussion by bilogic** [orchestral/testbench#405] [Q&A] [answered]
Is there a helper function to locate the fixture path inside a test? e.g. `fixture_path()`, just like base_path and storage_path

**Comment by crynobone** [comment, marked as answer]
Testbench recommends adding any fixtures to `workbench` directory, and you can use `workbench_path()`

**Reply by bilogic** [reply]
wonderful thanks!

**Reply by bilogic** [reply]
it's abit tedious to write `\Orchestra\Testbench\workbench_path()`, can have it as root namespace helper function?

**Reply by crynobone** [reply]
No plan to introduce global helper.

You should be able to import it:

```php
use function Orchestra\Testbench\workbench_path;
```

**Reply by bilogic** [reply]
```BASH
cd /laravel/project;
./vendor/bin/phpunit /laravel/project/vendor/my/package/tests/Feature/Example.php \
    --filter '^.*::test_example_case( .*)?$' \
    --configuration /laravel/project/vendor/my/package/phpunit.xml \
    --testdox \
    --stop-on-failure 
```

I use VSCode `BetterPHPUnit` to run my tests, the above is the CLI command it runs.
`workbench_path()` returns `/laravel/project/`, which is incorrect

Going to stick to my usual `__DIR__` first, but I think the `workbench_path()` and `package_path()` logic can be made better by detecting the "earliest" `composer.json` based on the test file's `__DIR__`

I can PR if ok

**Reply by bilogic** [reply]
```PHP
        $current = dirname(__DIR__);
        while (! file_exists("$current/composer.json")) {
            $current = dirname($current);
            if ($current == '/') {
                break;
            }
        }
        $package_path = $current;
        $workbench_path = "$package_path/tests/fixtures";
```
Is something like this acceptable?

**Comment by crynobone** [comment]
In latest version you should be able to do the following:

```bash
cd /laravel/project;
TESTBENCH_WORKING_PATH=/laravel/project/vendor/my/package php vendor/bin/phpunit /laravel/project/vendor/my/package/tests/Feature/Example.php \
    --filter '^.*::test_example_case( .*)?$' \
    --configuration /laravel/project/vendor/my/package/phpunit.xml \
    --testdox \
    --stop-on-failure 
```
